### PR TITLE
Generic client workers

### PR DESCRIPTION
**Pull request by raakella1**
Use separate grpc client worker pool for data and raft service

### CODE REVIEW

File: include/nuraft_mesg/mesg_factory.hpp [hunk 1]
```diff
@@ -38,16 +38,21 @@ using client_factory_lock_type = folly::SharedMutex;
...
     std::map< peer_id_t, std::shared_ptr< nuraft::rpc_client > > _clients;
 
 public:
     grpc_factory(int const cli_thread_count, std::string const& name);
+    grpc_factory(int const raft_cli_thread_count, int const data_cli_thread_count, std::string const& name);
     ~grpc_factory() override = default;
 
     std::string const& workerName() const { return _worker_name; }
+    std::string const& raftWorkerName() const { return _raft_worker_name; }
+    std::string const& dataWorkerName() const { return _data_worker_name; }
```
Comment (szmyd) [inline include/nuraft_mesg/mesg_factory.hpp:55]: Do we need to actually have two new member fields, can these accessories just return an inline `fmt` format around `workerName()` where it's called? It's just used at startup and shutdown yeah?

File: src/lib/manager_impl.cpp [hunk 2]
```diff
@@ -44,6 +41,12 @@ class engine_factory : public group_factory {
                                          start_params.ssl_cert_),
             application_(app) {}
```
Comment (szmyd) [inline src/lib/manager_impl.cpp:43]: Is the constructor above used anymore?

File: src/lib/nuraft_mesg_config.fbs [hunk 3]
```diff
@@ -15,6 +15,16 @@ table MesgFactoryConfig {
 
 table NuraftMesgConfig {
     mesg_factory_config: MesgFactoryConfig;
+
+    grpc_raft_client_thread_cnt: uint16 = 1;
+
+    grpc_data_client_thread_cnt: uint16 = 1;
+    
+    grpc_server_thread_cnt: uint16 = 2;
```
[the remark this answers is not in the document]
Comment (raakella1) [reply src/lib/nuraft_mesg_config.fbs:23]: yes, increased server thread cnt to 2 and a separate thread for data and raft channel clients

File: include/nuraft_mesg/mesg_factory.hpp (hunk 1 above)
Comment (raakella1) [reply include/nuraft_mesg/mesg_factory.hpp:55]: I was not sure if we wanted to remove the support to have a single threadpool for both channels. I was thinking of using whether these member fields are set as a proxy to the threading model. (if `_worker_name` is set then its a single thread pool etc based on the constructor used). If this is not necessary, I can just removethe support for a single thread pool

File: src/lib/manager_impl.cpp (hunk 2 above)
Comment (raakella1) [reply src/lib/manager_impl.cpp:43]: No, I wanted to ensure that we will never go back to the common threadpool before removing it completely.

File: include/nuraft_mesg/mesg_factory.hpp (hunk 1 above)
Comment (szmyd) [reply include/nuraft_mesg/mesg_factory.hpp:55]: But it's not functional code (single pool) today right? I just don't want it to never be needed and this code lives on forever. Your choice.

Comment (raakella1) [reply include/nuraft_mesg/mesg_factory.hpp:55]: Yeah, no one is using it. I am for removing it completely, I do not like having a code that we will not use. I put it out in the initial draft to see how it looks like. If @hkadayam is fine, I will remove it.

**Review by szmyd — APPROVED** [review]

**Pull request merged**